decomposer/trace_data.py

```diff
@@ -0,0 +1,28 @@
+# Copyright © 2025 Cognizant Technology Solutions Corp, www.cognizant.com.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# END COPYRIGHT
+
+from typing import Any
+
+
+class TraceData:
+    """
+    TraceData for a single MultiAgentReasoner
+    """
+
+    def __init__(self):
+        self.tree: dict[str, Any] = None
+        self.decomposition: dict[str, Any] = None
+        self.solve: dict[str, Any] = None
```

Comment:
Formal definition for TraceData as a class with member variables.
This avoids the loosey-goosey addition of fields that had previously been used with thread-local variables declared at the global scope.